### PR TITLE
Add HR dashboard widgets for workforce overview

### PR DESCRIPTION
**Pull request by patrickweh**
## Summary
- 17 widgets for the HR dashboard providing a company-wide workforce overview
- 6 ValueBox KPIs: headcount, absences today, total overtime, vacation balance, sick rate, new hires
- 6 Charts: work time comparison, absence type distribution, sick days trend, department headcount, work time model distribution, overtime trend
- 5 Custom widgets: upcoming absences, top overtime ranking, upcoming birthdays, active work times, vacation heatmap
- 11 default widgets configured on the dashboard, all 17 available for manual addition
- 45 tests covering rendering and business logic

## Summary by Sourcery

Add a time-frame-aware HR dashboard with a suite of workforce overview widgets and default layout configuration.

New Features:
- Introduce headcount, absences today, overtime total, vacation balance, sick rate, and new hires KPI widgets for the HR dashboard.
- Add charts for work time comparison, absence type distribution, department headcount, work time model distribution, overtime trend, and sick days trend.
- Provide custom HR widgets for upcoming absences, top overtime employees, upcoming birthdays, active work times, and a vacation heatmap, including corresponding Blade views.
- Enable time range selection on the HR dashboard and register a default set of HR widgets via the widget facade.

Tests:
- Add comprehensive Livewire and model interaction tests covering rendering and business logic for all new HR dashboard widgets, including edge cases with no data.

### CODE REVIEW

File: src/Livewire/HumanResources/Dashboard.php [hunk 1]
```diff
@@ -2,16 +2,40 @@
...
 class Dashboard extends BaseDashboard
 {
-    protected bool $hasTimeSelector = false;
+    protected bool $hasTimeSelector = true;
 
     public static function getDefaultWidgets(): array
     {
         return parent::mapDefaultWidgets(
-            static::$defaultWidgets ?? []
+            static::$defaultWidgets ?? [
```
Comment (SirSplasch) [inline src/Livewire/HumanResources/Dashboard.php:27]: Can/Should the one currently existing HR dashboard widget be added to the default?

File: src/Livewire/Widgets/HumanResources/AbsencesTodayBox.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public function calculateSum(): void
+    {
+        $today = now()->toDateString();
+
+        $absentCount = resolve_static(AbsenceRequest::class, 'query')
+            ->where('state', AbsenceRequestStateEnum::Approved)
+            ->where('start_date', '<=', $today)
+            ->where('end_date', '>=', $today)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/AbsencesTodayBox.php]: whereValueBetween() instead

File: src/Livewire/Widgets/HumanResources/AbsencesTodayBox.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            ->distinct('employee_id')
+            ->count('employee_id');
+
+        $totalActive = resolve_static(Employee::class, 'query')
+            ->employed(now())
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/AbsencesTodayBox.php]: Do you need to use now() or can you use $today?

File: src/Livewire/Widgets/HumanResources/AbsenceTypeDistributionChart.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                $query->whereBetween('date', [$startDate, $endDate]);
+            })
+            ->with('absenceType:id,name,color')
+            ->get(['id', 'absence_type_id', 'work_days_affected']);
+
+        $absenceByType = [];
+        foreach ($absenceRequests as $absenceRequest) {
+            $absenceTypeId = $absenceRequest->absence_type_id;
+
+            if (! isset($absenceByType[$absenceTypeId])) {
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/AbsenceTypeDistributionChart.php]: ! array_key_exists()

File: src/Livewire/Widgets/HumanResources/AbsenceTypeDistributionChart.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                2
+            );
+        }
+
+        $labels = [];
+        $series = [];
+        $colors = [];
+
+        foreach ($absenceByType as $typeData) {
+            if (bccomp($typeData['days'], 0, 2) > 0) {
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/AbsenceTypeDistributionChart.php]: bccomp($typeData['days'], 0) === 1

File: src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return 4;
+    }
+
+    public static function getDefaultWidth(): int
+    {
+        return 2;
+    }
+
+    public function calculateList(): void
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php:41]: WorkTime does not necessarily have a user. The work time can have an employee instead.

File: src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                           </span>'
+                    )
+                    . '<div>' . $item->user?->name . '</div>
+                </div>',
+            'value' => $item->started_at
+                ->locale(app()->getLocale())
+                ->timezone(auth()->user()?->timezone ?? config('app.timezone'))
+                ->isoFormat('L LT'),
+            'growthRate' => null,
+        ])->toArray();
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php]: wrap
->toArray();

File: src/Livewire/Widgets/HumanResources/NewHiresBox.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return 5;
+    }
+
+    public static function getDefaultOrderRow(): int
+    {
+        return 0;
+    }
+
+    #[Renderless]
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/NewHiresBox.php:36]: missing import, as the ai pointed out

File: src/Livewire/Widgets/HumanResources/NewHiresBox.php [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        $newHires = resolve_static(Employee::class, 'query')
+            ->whereBetween('employment_date', [$start, $end])
+            ->count();
+
+        $departures = resolve_static(Employee::class, 'query')
+            ->whereBetween('termination_date', [$start, $end])
+            ->count();
+
+        $this->sum = Number::format($newHires);
+        $this->subValue = __(':count departures', ['count' => $departures]);
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/NewHiresBox.php]: I'm a little confused as to why the departures are shown in the "NewHires" widget.

File: src/Livewire/Widgets/HumanResources/SickRateBox.php [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return 4;
+    }
+
+    public static function getDefaultOrderRow(): int
+    {
+        return 0;
+    }
+
+    #[Renderless]
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/SickRateBox.php:38]: missing import

File: src/Livewire/Widgets/HumanResources/SickRateBox.php [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    public function calculateSum(): void
+    {
+        $start = $this->getStart();
+        $end = $this->getEnd();
+
+        $currentQuery = resolve_static(EmployeeDay::class, 'query')
+            ->where('is_work_day', true)
+            ->whereBetween('date', [$start, $end]);
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/SickRateBox.php]: swap these two, to match table column ordering

File: src/Livewire/Widgets/HumanResources/SickRateBox.php [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+        $this->calculateSum();
+    }
+
+    public function calculateSum(): void
+    {
+        $start = $this->getStart();
+        $end = $this->getEnd();
+
+        $currentQuery = resolve_static(EmployeeDay::class, 'query')
+            ->where('is_work_day', true)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/SickRateBox.php]: You are constraining this query by is_work_day, but in the SickDaysTrendChart you are not using this constraint. This seems inconsistent to me.

File: src/Livewire/Widgets/HumanResources/SickRateBox.php [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        $this->subValue = __(':days sick days total', [
+            'days' => Number::format(bcround($sickDaysSum, 1), 1),
+        ]);
+
+        $previousStart = $this->getStartPrevious();
+        $previousEnd = $this->getEndPrevious();
+
+        $previousQuery = resolve_static(EmployeeDay::class, 'query')
+            ->where('is_work_day', true)
+            ->whereBetween('date', [$previousStart, $previousEnd]);
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/SickRateBox.php]: swap

File: src/Livewire/Widgets/HumanResources/TopOvertimeWidget.php [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+use FluxErp\Traits\Livewire\Widget\Widgetable;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Number;
+use Livewire\Component;
+
+class TopOvertimeWidget extends Component
+{
+    use Widgetable;
+
+    public array $employees = [];
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/TopOvertimeWidget.php:18]: Add #[Locked] to property?

File: src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+use FluxErp\Models\AbsenceRequest;
+use FluxErp\Traits\Livewire\Widget\Widgetable;
+use Illuminate\Contracts\View\View;
+use Livewire\Component;
+
+class UpcomingAbsencesWidget extends Component
+{
+    use Widgetable;
+
+    public array $absences = [];
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php:21]: Add #[Locked] to property?

File: src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $this->loadAbsences();
+    }
+
+    public function render(): View
+    {
+        return view('flux::livewire.widgets.human-resources.upcoming-absences');
+    }
+
+    public function loadAbsences(): void
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php]: Why did you derive from the default? I think the common method name is "loadData"

File: src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $today = now()->toDateString();
+        $twoWeeksFromNow = now()->addDays(14)->toDateString();
+
+        $this->absences = resolve_static(AbsenceRequest::class, 'query')
+            ->where('state', AbsenceRequestStateEnum::Approved)
+            ->where('end_date', '>=', $today)
+            ->where('start_date', '<=', $twoWeeksFromNow)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php]: swap

File: src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+            ->with(['employee:id,name', 'absenceType:id,name,color'])
+            ->orderBy('start_date')
+            ->limit(15)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php]: Why is this limited to 15 absences?

File: src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+            ->get()
+            ->map(fn (AbsenceRequest $absence) => [
+                'employee_name' => $absence->employee?->name,
+                'absence_type' => $absence->absenceType?->name,
+                'color' => $absence->absenceType?->color ?? '#6b7280',
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingAbsencesWidget.php]: Can't you use the ChartColorEnum to define the fallback color?

File: src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+use FluxErp\Models\Employee;
+use FluxErp\Traits\Livewire\Widget\Widgetable;
+use Illuminate\Contracts\View\View;
+use Livewire\Component;
+
+class UpcomingBirthdaysWidget extends Component
+{
+    use Widgetable;
+
+    public array $birthdays = [];
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php:20]: Add #[Locked] to property?

File: src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        $this->loadBirthdays();
+    }
+
+    public function render(): View
+    {
+        return view('flux::livewire.widgets.human-resources.upcoming-birthdays');
+    }
+
+    public function loadBirthdays(): void
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php]: loadData?

File: src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+                return [
+                    'name' => $employee->name,
+                    'date' => $birthday->locale(app()->getLocale())->isoFormat('D. MMM'),
+                    'date_sort' => $birthday->toDateString(),
+                    'age' => $age,
+                ];
+            })
+            ->filter()
+            ->sortBy('date_sort')
+            ->take(10)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php]: Why only take 10? If more than 10 employees have an upcoming birthday they will be ignored.

File: src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+            ->map(function (Employee $employee) use ($today, $endDate) {
+                $birthday = $employee->date_of_birth->copy()->year($today->year);
+
+                if ($birthday->lt($today)) {
+                    $birthday->addYear();
+                }
+
+                if ($birthday->gt($endDate)) {
+                    return null;
+                }
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php]: Why don't you filter the employees date_of_birth by sql?
If you need to know how, you can look at the scopeInTimeframe method of the Address model.

File: src/Livewire/Widgets/HumanResources/VacationBalanceTotalBox.php [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+    public static function getDefaultOrderRow(): int
+    {
+        return 0;
+    }
+
+    public function calculateSum(): void
+    {
+        $employees = resolve_static(Employee::class, 'query')
+            ->employed(now())
+            ->select(['id', 'employment_date', 'termination_date', 'is_active', 'vacation_carryover_rule_id'])
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationBalanceTotalBox.php]: order columns by table

File: src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php [hunk 25]
```diff
@@ -0,0 +1,130 @@
...
+use FluxErp\Models\Employee;
+use FluxErp\Traits\Livewire\Widget\Widgetable;
+use Illuminate\Contracts\View\View;
+use Livewire\Component;
+
+class VacationHeatmapWidget extends Component
+{
+    use Widgetable;
+
+    public array $weeks = [];
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php:20]: Add #[Locked] to property?

File: src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        $this->loadHeatmap();
+    }
+
+    public function render(): View
+    {
+        return view('flux::livewire.widgets.human-resources.vacation-heatmap');
+    }
+
+    public function loadHeatmap(): void
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php]: loadData?

File: src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php [hunk 27]
```diff
@@ -0,0 +1,130 @@
...
+        $endDate = $today->copy()->addWeeks(6)->endOfWeek();
+
+        $totalActive = resolve_static(Employee::class, 'query')
+            ->employed(now())
+            ->count();
+
+        $absences = resolve_static(AbsenceRequest::class, 'query')
+            ->where('state', AbsenceRequestStateEnum::Approved)
+            ->where('end_date', '>=', $today->toDateString())
+            ->where('start_date', '<=', $endDate->toDateString())
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php]: swap

File: src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php [hunk 28]
```diff
@@ -0,0 +1,130 @@
...
+            ->get(['id', 'employee_id', 'start_date', 'end_date']);
+
+        $absentCountByDate = [];
+        foreach ($absences as $absence) {
+            $current = $absence->start_date->copy()->max($today);
+            $end = $absence->end_date->copy()->min($endDate);
+
+            while ($current->lte($end)) {
+                $dateKey = $current->toDateString();
+                $absentCountByDate[$dateKey] = ($absentCountByDate[$dateKey] ?? collect())->push($absence->employee_id);
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php]: The only thing this does is fill the variable with a single item collection with the employee_id of the current absence, if the key does not exist.
I suppose this cannot be right, as you later make a unique()->count() on this collection, but it only ever holds one item and therefore ever only returns 1.

File: src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php [hunk 29]
```diff
@@ -0,0 +1,130 @@
...
+    public function loadHeatmap(): void
+    {
+        $today = now()->startOfDay();
+        $endDate = $today->copy()->addWeeks(6)->endOfWeek();
+
+        $totalActive = resolve_static(Employee::class, 'query')
+            ->employed(now())
+            ->count();
+
+        $absences = resolve_static(AbsenceRequest::class, 'query')
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationHeatmapWidget.php:76]: This widget suggests that this is a heatmap over the vacations, so why do you not filter the absence requests so that only vacations are used?

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+        return 2;
+    }
+
+    public function calculateChart(): void
+    {
+        $now = now();
+
+        $employedIds = resolve_static(Employee::class, 'query')
+            ->employed($now)
+            ->select('id')
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: no need to add the select() if you pluck the query

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 31]
```diff
@@ -0,0 +1,105 @@
...
+            $this->labels = [];
+
+            return;
+        }
+
+        $workTimeModelTable = resolve_static(WorkTimeModel::class, 'query')->getModel()->getTable();
+        $pivotTable = resolve_static(EmployeeWorkTimeModel::class, 'query')->getModel()->getTable();
+
+        $distribution = resolve_static(EmployeeWorkTimeModel::class, 'query')
+            ->whereIn($pivotTable . '.employee_id', $employedIds)
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: whereIntegerInRaw

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 32]
```diff
@@ -0,0 +1,105 @@
...
+
+        if ($employedIds->isEmpty()) {
+            $this->series = [];
+            $this->labels = [];
+
+            return;
+        }
+
+        $workTimeModelTable = resolve_static(WorkTimeModel::class, 'query')->getModel()->getTable();
+        $pivotTable = resolve_static(EmployeeWorkTimeModel::class, 'query')->getModel()->getTable();
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: Like isn't it easier to do app(Model::class)->getTable()?

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+
+        $distribution = resolve_static(EmployeeWorkTimeModel::class, 'query')
+            ->whereIn($pivotTable . '.employee_id', $employedIds)
+            ->where($pivotTable . '.valid_from', '<=', $now)
+            ->where(fn ($query) => $query->whereNull($pivotTable . '.valid_until')
+                ->orWhere($pivotTable . '.valid_until', '>=', $now))
+            ->join($workTimeModelTable, $pivotTable . '.work_time_model_id', '=', $workTimeModelTable . '.id')
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: join before where constraints

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 34]
```diff
@@ -0,0 +1,105 @@
...
+            ->selectRaw($workTimeModelTable . '.name, ' . $workTimeModelTable . '.id, count(' . $pivotTable . '.employee_id) as employee_count')
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: maximum line length exceeded

File: src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php [hunk 35]
```diff
@@ -0,0 +1,105 @@
...
+        $colors = [];
+
+        foreach ($distribution as $index => $row) {
+            $labels[] = $row->name;
+            $series[] = (int) $row->employee_count;
+            $colors[] = resolve_static(
+                ChartColorEnum::class,
+                'forIndex',
+                ['index' => $index]
+            )->value;
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/WorkTimeModelDistributionChart.php]: wrap
->value();

**Review by SirSplasch — CHANGES_REQUESTED** [review]
Add missing translations

File: src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php [hunk 36]
```diff
@@ -0,0 +1,90 @@
...
+            ->where('is_locked', false)
+            ->where('is_pause', false)
+            ->with(['user:id,name', 'employee:id,name'])
+            ->get();
+
+        $usersWithPause = resolve_static(WorkTime::class, 'query')
+            ->where('is_daily_work_time', true)
+            ->where('is_locked', false)
+            ->where('is_pause', true)
+            ->whereIn('user_id', $workTimes->pluck('user_id'))
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/ActiveWorkTimesWidget.php]: whereIntegerInRaw() and put it as the first constraint, to match table ordering.

File: src/Livewire/Widgets/HumanResources/HeadcountBox.php [hunk 37]
```diff
@@ -0,0 +1,63 @@
...
+
+        $newHires = resolve_static(Employee::class, 'query')
+            ->whereBetween('employment_date', [$monthStart, $monthEnd])
+            ->count();
+
+        $departures = resolve_static(Employee::class, 'query')
+            ->whereBetween('termination_date', [$monthStart, $monthEnd])
+            ->count();
+
+        $this->subValue = __(':hires / :departures this month', [
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/HeadcountBox.php:53]: translation missing in de.json

File: src/Livewire/Widgets/HumanResources/OvertimeTotalBox.php [hunk 38]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        $employeeCount = $employees->count();
+        $this->sum = Number::format(bcround($totalOvertime, 2), 2) . 'h';
+
+        $average = $employeeCount > 0
+            ? bcround(bcdiv($totalOvertime, $employeeCount, 4), 2)
+            : '0.00';
+
+        $this->subValue = __('Ø :hours h per employee', [
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/OvertimeTotalBox.php:53]: translation missing in de.json

File: lang/de.json [hunk 39]
```diff
@@ -2386,12 +2409,17 @@
...
+    "Sick Days Trend": "Krankheitstage-Verlauf",
+    "Sick Days Trend Chart": "Krankheitstage-Verlauf",
     "Sick Days Used": "Genutzte Krankheitstage",
     "Sick days used and vacation days used cannot exceed 1 day in total.": "Krankheitstage und genutzte Urlaubstage dürfen insgesamt 1 Tag nicht überschreiten.",
     "Sick Hours Used": "Genutzte Krankheitsstunden",
     "Sick hours used and vacation hours used cannot exceed 24 hours in total.": "Krankheitsstunden und genutzte Urlaubsstunden dürfen insgesamt 24 Stunden nicht überschreiten.",
     "Sick Leave": "Krankmeldung",
     "Sick Note Issued Date": "Krankmeldung ausgestellt am",
+    "Sick Rate": "Krankenquote",
+    "Sick Rate Box": "Krankenquote",
```
Comment (SirSplasch) [inline lang/de.json]: What about "Krankheitsquote"?

File: src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php [hunk 40]
```diff
@@ -0,0 +1,104 @@
...
+
+                return [
+                    'name' => $employee->name,
+                    'date' => $birthday->locale(app()->getLocale())->isoFormat('D. MMM'),
+                    'date_sort' => $birthday->toDateString(),
+                    'age' => $employee->date_of_birth->diffInYears($birthday),
+                ];
+            })
+            ->sortBy('date_sort')
+            ->values()
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/UpcomingBirthdaysWidget.php]: I don't think you need to do ->values() as you have not touched the indexes.

File: src/Livewire/Widgets/HumanResources/VacationBalanceTotalBox.php [hunk 41]
```diff
@@ -0,0 +1,62 @@
...
+        $employeeCount = $employees->count();
+
+        $this->sum = Number::format(bcround($totalBalance, 1), 1)
+            . ' ' . __('days');
+
+        $average = $employeeCount > 0
+            ? bcround(bcdiv($totalBalance, $employeeCount, 4), 1)
+            : '0.0';
+
+        $this->subValue = __('Ø :days days per employee', [
```
Comment (SirSplasch) [inline src/Livewire/Widgets/HumanResources/VacationBalanceTotalBox.php:53]: translation missing in de.json

File: lang/de.json [hunk 42]
```diff
@@ -2774,7 +2808,10 @@
     "Uses vacation days balance": "Verwendet Urlaubskontingent",
     "Uuid": "UUID",
     "Vacation": "Urlaub",
+    "Vacation Balance": "Urlaubssaldo",
     "Vacation Balance Box": "Urlaubskontostand",
+    "Vacation Balance Total": "Urlaubssaldo gesamt",
+    "Vacation Balance Total Box": "Urlaubssaldo gesamt",
```
Comment (SirSplasch) [inline lang/de.json]: "Urlaubssaldo" vs. "Urlaubskontostand"

File: resources/views/livewire/widgets/human-resources/vacation-heatmap.blade.php [hunk 43]
```diff
@@ -0,0 +1,76 @@
+<div class="flex h-full flex-col gap-4 p-4">
+    <div>
+        <h2
+            class="truncate text-lg font-semibold text-gray-700 dark:text-gray-400"
+        >
+            {{ __('Absence Heatmap') }}
```
Comment (SirSplasch) [inline resources/views/livewire/widgets/human-resources/vacation-heatmap.blade.php]: This key should be changed to 'Vacation Heatmap' to match the widget's name.

File: lang/de.json [hunk 44]
```diff
@@ -53,6 +55,7 @@
     "A token with this instance name already exists. Regenerating will invalidate the old token. Do you want to continue?": "Ein Token mit diesem Instanznamen existiert bereits. Die Neugenerierung wird den alten Token ungültig machen. Möchten Sie fortfahren?",
     "Abbreviation": "Abkürzung",
     "Abilities": "Fähigkeiten",
+    "Absence Heatmap": "Abwesenheits-Heatmap",
```
Comment (SirSplasch) [inline lang/de.json]: Change this to "Vacation Heatmap": "Urlaubs-Heatmap" and order it accordingly.

**Review by SirSplasch — CHANGES_REQUESTED** [review]
Annotations to WorkTimeModelDistributionChart were ignored in the previous changes.

**Review by SirSplasch — APPROVED** [review]
